ocfweb/lab_reservations/reserve.py

```diff
@@ -0,0 +1,178 @@
+from collections import namedtuple
+
+import ocflib.misc.validators
+from django import forms
+from django.shortcuts import render
+from jinja2 import Environment
+from jinja2 import PackageLoader
+from ocflib.account.search import user_attrs_ucb
+from ocflib.misc.mail import send_mail
+
+from ocfweb.auth import calnet_required
+from ocfweb.component.forms import wrap_validator
+
+
+JINJA_MAIL_ENV = Environment(loader=PackageLoader('ocfweb', ''))
+
+
+class NewReservationRequest(namedtuple(
+    'NewReservationRequest', [
+        'real_name',
+        'contact_email',
+        'student_group',
+        'reason',
+        'date',
+        'starttime',
+        'endtime',
+    ],
+)):
```

Comment:
If you're going to extend a nametuple, it's best practice to set `__slots__ = ()` again in order to make it immutable.

Here's why:

```python
# Not using inheritence
>>> Point = collections.namedtuple('Point', ('x', 'y'))
>>> p = Point(x=32, y=16)
>>> p.z = 14
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'Point' object has no attribute 'z'
# ^-- good, can't jam arbitrary attributes onto it

# Trying again, this time using inheritence
>>> class Point(collections.namedtuple('Point', ('x', 'y'))):
...     pass
... 
>>> p = Point(x=32, y=16)
>>> p.z = 14
>>>
# ^-- bad, can jam arbitrary attributes on it (and mutate them later!)

# The fix is to explicitly set the slots:
>>> class Point(collections.namedtuple('Point', ('x', 'y'))):
...     __slots__ = ()
... 
>>> p = Point(x=32, y=16)
>>> p.z = 14
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'Point' object has no attribute 'z'
# ^-- good, can't jam arbitrary attributes onto it again
```

Reply:
still need to do this?